server/server/Models/order.js

```diff
@@ -103,12 +103,24 @@ const orderSchema = new mongoose.Schema(
   },
 );
 
-// order document will be deleted after 1 hour only if order status was still delivering
+// order document will be deleted after 1 hour only if order status was still delivering or cancelled
 orderSchema.index(
   { createdAt: 1 },
   {
     expireAfterSeconds: 3600,
-    partialFilterExpression: { orderStatus: "delivering" },
+    partialFilterExpression: {
+      orderStatus: "delivering",
+    },
+  },
+);
+
+orderSchema.index(
+  { orderDate: 1 },
+  {
+    expireAfterSeconds: 3600,
+    partialFilterExpression: {
+      orderStatus: "cancelled",
```

Comment:
well done, If you want you can combine `orderStatus ` into a single index using the `$in` operator

Reply:
yeah i've tried that but it is not supported with our mongoDB docker version `5`, it should be version `+6`

Reply:
![image](https://github.com/202309-EKTA-JO-FSW/capstone-room-5/assets/80969606/69b1eed2-1659-4c71-9957-2ce5c1bdc024)
this is supported in version `7`

![image](https://github.com/202309-EKTA-JO-FSW/capstone-room-5/assets/80969606/f2042e07-70ef-4b1d-97f8-3bca35f9f1fa)
sadly it's not supported in our version, I've tried `$and`, but didn't work